### PR TITLE
FXIOS-1808: Reading list items in recentlySaved - iphone only

### PR DESCRIPTION
**Pull request by nishant2718**
This PR addresses [this JIRA ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-1808). Please see [the design specs](https://www.figma.com/file/am9EJGg6ow29PXWkc9jcOH/MR2.1?node-id=0%3A1) to verify. 

Remaining items:
- Verification by a Data Steward.

### CODE REVIEW

**Comment by nishant2718** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) What questions will you answer with this data?

    - The goal of this metric is to understand if users are saving reading list items enough to make this feature valuable, so we want to know how often the recently saved section displays them, and the number of reading list items.

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:
    - These events will help us understand the usage of the recently saved section, so we can confirm its value for users.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?

   None

4) Can current instrumentation answer these questions?

   No

5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.   
Category 2

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>

  <tr>
    <td>The number of reading list items displayed on the homepage is tracked in Glean</td>
    <td>Category 2</td>
    <td>https://jira.mozilla.com/browse/FXIOS-1808</td>
  </tr>
  <tr>


  <tr>
    <td>The number of clicks on the reading list item from the home screen is tracked in Glean</td>
    <td>Category 2</td>
    <td>https://jira.mozilla.com/browse/FXIOS-1808</td>
  </tr>
  <tr>

  <tr>
</table>

6) Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
Available in metrics.yaml and https://dictionary.telemetry.mozilla.org/apps/firefox_ios

7) How long will this data be collected?  Choose one of the following:

    Until 2022-01-01 (~6 months) so we can gather enough data

8) What populations will you measure?
All

9) If this data collection is default on, what is the opt-out mechanism for users?
Initially it'll be hidden behind a feature flag, and not on by default. You can toggle this section on/off via the debug menu.

10) Please provide a general description of how you will analyze this data.
Using Glean / Amplitude

11) Where do you intend to share the results of your analysis?
Only on Glean, Amplitude and with mobile teams.

12) Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection? If so:
None

**Comment by dnarcese** [comment]
r+
# Data Review Form

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, telemetry is documented in `docs/metrics.md`

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, Firefox iOS provides data controls in the app settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

This should be re-reviewed w/ Ashley Thomas every 6mo

4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Type 2

5) Is the data collection request for default-on or default-off?

Default on

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does the data collection use a third-party collection tool? 

No

**Review by dnarcese — APPROVED** [review]

File: Client/RecentItemsHelper.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Storage
+
+class RecentItemsHelper {
+    
+    /// Filter items that are a older than some number of days.
```
Comment (electricRGB) [inline Client/RecentItemsHelper.swift:10]:
```suggestion
    /// Filter `RecentlySavedItem`s that are older than a given number of days.
```

File: Client/Frontend/Home/FxHomeRecentlySavedCollectionCell.swift [hunk 2]
```diff
@@ -131,7 +170,7 @@ extension FxHomeRecentlySavedCollectionCell: UICollectionViewDelegateFlowLayout
     
 }
 
-private struct RecentlySavedBookmarkCellUX {
+private struct RecentlySavedCellUX {
```
Comment (electricRGB) [inline Client/Frontend/Home/FxHomeRecentlySavedCollectionCell.swift:173]: 👍

**Review by electricRGB — APPROVED** [review]

File: Client/RecentItemsHelper.swift (hunk 1 above)
Comment (nishant2718) [reply Client/RecentItemsHelper.swift:10]: I'll fix this in the iPad PR 😬 Should be coming shortly after

**Pull request merged**